nexus/mgs-updates/src/driver_update.rs

```diff
@@ -349,16 +362,33 @@ pub(crate) async fn apply_update(
 
     if try_reset {
         // We retry this until we get some error *other* than a communication
-        // error.  There is intentionally no timeout here.  If we've staged an
-        // update but not managed to reset the device, there's no point where
-        // we'd want to stop trying to do so.
+        // error or an RoT bootloader image error.  There is intentionally no
+        // timeout here.  If we've staged an update but not managed to reset
+        // the device, there's no point where we'd want to stop trying to do so.
         while let Err(error) =
             update_helper.post_update(log, &mut mgs_clients, update).await
         {
-            if !matches!(error, gateway_client::Error::CommunicationError(_)) {
-                let error = InlineErrorChain::new(&error);
-                error!(log, "post_update failed"; &error);
-                return Err(ApplyUpdateError::SpResetFailed(error.to_string()));
+            match error {
+                PostUpdateError::GatewayClientError(error) => {
```

Comment:
suggestion: add a `is_transient()` (or `is_fatal()`) to `PostUpdateError`.  Then replace this whole match block with:

```rust
if !error.is_transient() {
    let error = InlineErrorChain::new(&error);    
    error!(log, "post_update failed"; &error);
    return Err(ApplyUpdateError::SpResetFailed(
        error.to_string(),
    ));
}
```